craft_parts/plugins/rust_plugin.py

```diff
@@ -35,13 +36,20 @@
 
 logger = logging.getLogger(__name__)
 
+
+def _unique_str_list(v: List[str]) -> List[str]:
+    if len(v) != len(set(v)):
+        raise ValueError("not unique string list")
+    return v
+
+
 # A workaround for mypy false positives
 # see https://github.com/samuelcolvin/pydantic/issues/975#issuecomment-551147305
 # The proper fix requires Python 3.9+ (needs `typing.Annotated`)
 if TYPE_CHECKING:
     UniqueStrList = List[str]
 else:
-    UniqueStrList = conlist(str, unique_items=True)
+    UniqueStrList = Annotated[List[str], AfterValidator(_unique_str_list)]
```

Comment:
Yeah I think Set will be fit-in replacement for UniqueStrList.

Reply:
After reconsideration, `Set` may not be a good fit because order may be important in some fields like `rust_path`.
As for a generic `UniqueList`, there isn't a reference for another type of unique lists used in the project, so I guess it is not necessary for now.

Reply:
Can you remove the mypy workaround? It seems to be fixed now.

Reply:
done